core/src/main/java/io/kestra/core/runners/pebble/functions/KvFunction.java

```diff
@@ -0,0 +1,76 @@
+package io.kestra.core.runners.pebble.functions;
+
+import io.kestra.core.exceptions.IllegalVariableEvaluationException;
+import io.kestra.core.exceptions.ResourceExpiredException;
+import io.kestra.core.runners.RunVariables;
+import io.kestra.core.secret.SecretService;
+import io.kestra.core.services.FlowService;
+import io.kestra.core.storages.StorageInterface;
+import io.kestra.core.storages.kv.InternalKVStore;
+import io.kestra.core.utils.TruthUtils;
+import io.pebbletemplates.pebble.error.PebbleException;
+import io.pebbletemplates.pebble.extension.Function;
+import io.pebbletemplates.pebble.template.EvaluationContext;
+import io.pebbletemplates.pebble.template.PebbleTemplate;
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Consumer;
+
+@Slf4j
+@Singleton
+public class KvFunction implements Function {
+    @Inject
+    private FlowService flowService;
+
+    @Inject
+    private StorageInterface storageInterface;
```

Comment:
Should not be dependent of the StorageInterface, I think we miss a KvStoreService to hide the implementation.